engine/baml-cli/src/version_command.rs

```diff
@@ -0,0 +1,263 @@
+use crate::{builder::get_src_dir, shell::build_shell_command};
+use crate::{command::run_command_with_error, errors::CliError, OutputType};
+use clap;
+use colored::Colorize;
+use log;
+use regex::Regex;
+use serde;
+use std::path::PathBuf;
+
+#[derive(Debug, serde::Deserialize)]
+pub struct LatestVersionsManifest {
+    pub cli: Option<String>,
+    pub py_client: Option<String>,
+    pub ts_client: Option<String>,
+    pub vscode: Option<String>,
+}
+
+#[derive(Debug, serde::Serialize)]
+pub struct CheckedVersions {
+    pub cli: CliVersion,
+    pub clients: Vec<ClientVersions>,
+    pub vscode: VscodeVersion,
+}
+
+#[derive(Debug, serde::Serialize)]
+pub struct CliVersion {
+    pub current_version: String,
+    pub latest_version: Option<String>,
+}
+
+#[derive(Debug, serde::Serialize)]
+pub struct ClientVersions {
+    pub dir: PathBuf,
+    pub language: String,
+    pub current_version: String,
+    pub latest_version: Option<String>,
+}
+
+#[derive(Debug, serde::Serialize)]
+pub struct VscodeVersion {
+    pub latest_version: Option<String>,
+}
+
+pub fn get_client_version(
+    project_root: &str,
+    package_version_command: &str,
+) -> Result<String, CliError> {
+    let cmd = shellwords::split(package_version_command)
+        .map_err(|e| CliError::StringError(e.to_string()))?;
+
+    let mut cmd = build_shell_command(cmd);
+
+    // NOTE(sam): no idea why this has to start in the cwd; this is copied from update_client.rs
+    // according to vbv@ this had to be done for _some_ reason, so just preserving it as closely as i can
+    let cwd = std::env::current_dir()?.canonicalize()?;
```

Comment:
yes theres some weird shit that goes on on Windows vs MacOS with this. I appreciate the comment!